awsiot/iotshadow.py

```diff
@@ -1323,10 +1338,19 @@ def from_payload(cls, payload):
     def to_payload(self):
```

Comment:
this "explicitly null" thing can go both ways. Users can send AND receive `ShadowState`.

We should also adjust the `from_payload()` function so that users can tell the difference between a "None because nothing was there" and "None because the json doc had null in it"

In my opinion, the naming scheme `xyz_none_is_valid` makes a little less sense when we think about data being received. maybe we should just go with something simple like `xyz_is_explicitly_null` and if that's true, then we just send null. We don't cross-reference it with the actual value of `xyz`

Reply:
That works for me - that would also make the code simpler as well.

Reply:
I changed the name to `<var>_is_nullable` and added checking to `from_payload` 👍